msteams-platform/toolkit/TeamsFx-SDK.md

```diff
@@ -402,21 +344,268 @@ The `@microsoft/mgt-teamsfx-provider` package exposes the `TeamsFxProvider` clas
+      // Query user's information from the access token.
+      try {
+        const currentUser: UserInfo = await teamsfx.getUserInfo();
+        if (currentUser && currentUser.displayName) {
+          res.body.userInfoMessage = `User display name is ${currentUser.displayName}.`;
+        } else {
+          res.body.userInfoMessage = "No user information was found in access token.";
+        }
+      } catch (e) {
+      }
+      // Create a graph client to access user's Microsoft 365 data after user has consented.
+      try {
+        const graphClient: Client = createMicrosoftGraphClient(teamsfx, [".default"]);
+        const profile: any = await graphClient.api("/me").get();
+        res.body.graphClientMessage = profile;
+      } catch (e) {
+      }
+      return res;
+    }
+    ```
 
-<br>
+    For more information on sample to use graph API in bot application, see  [hello-world-tab-with-backend sample](https://github.com/OfficeDev/TeamsFx-Samples/tree/dev/hello-world-tab-with-backend).
+
+    </details>
+
+  * Application permission in backend
+    <details>
+    <summary><b>Use certificate-based authentication in Azure Function</b></summary>
+
+    This code snippet shows you how to use certificate-based application permission to get the token that can be used to call Graph API.
+
+    1. You can initialize the `authConfig` by providing a `PEM-encoded key certificate`.
+
+    ```ts
+    const authConfig = {
+      clientId: process.env.M365_CLIENT_ID,
+      certificateContent: "The content of a PEM-encoded public/private key certificate",
+      authorityHost: process.env.M365_AUTHORITY_HOST,
+      tenantId: process.env.M365_TENANT_ID,
+    };
+    ```
+
+    2. You can use the `authConfig` to get the token.
+
+    ```ts
+    const teamsfx = new TeamsFx(IdentityType.App);
+    teamsfx.setCustomeConfig(authConfig);
+    const token = teamsfx.getCredential().getToken();
+    ```
+
+    </details>
+
+    <details>
+    <summary><b>Use client secret authentication in Azure Function</b></summary>
+
+    This code snippet shows you how to use client secret application permission to get the token that can be used to call Graph API.
+
+    1. You can initialize the `authConfig` by providing a `client secret`.
+
+    ```ts
+    const authConfig = {
+      clientId: process.env.M365_CLIENT_ID,
+      clientSecret: process.env.M365_CLIENT_SECRET,
+      authorityHost: process.env.M365_AUTHORITY_HOST,
+      tenantId: process.env.M365_TENANT_ID,
+    };
+    ```
+
+    2. You can use the `authConfig` to get the token.
+
+    ```ts
+    const teamsfx = new TeamsFx(IdentityType.App);
+    teamsfx.setCustomeConfig(authConfig);
+    const token = teamsfx.getCredential().getToken();
+    ```
+
+    For more information on sample to use graph API in bot application, see the [hello-world-tab-with-backend sample](https://github.com/OfficeDev/TeamsFx-Samples/tree/dev/hello-world-tab-with-backend).
+
+    </details>
+
+* Other scenario:
+  <details>
```

Comment:
I feel like `Other scenarios` shouldn't be under `Microsoft Graph Scenarios` section?

Reply:
This can be done.